### ISSUE TITLE
Select EBCDIC graphic data returns java.io.CharConversionException, ERRORCODE=-4220

### ISSUE THREAD

**Issue by pymak7** [IBM/db2forzosdeveloperextension-about#81]
## Development environment where the bug occurred

- Db2 Developer Extension version: 1.3.5
- Editor platform
  - [x] Visual Studio Code
  - [ ] Eclipse Theia
- Editor platform version: 1.66.2
- Operating system on which your editor runs (for example, Windows 10 2004 or MacOS Catalina 10.15.7): MacOS Monterey 12.3.1
- Java Version (Run `java -version` and paste the details here): 1.8.0_311
- Db2 for z/OS version (including function level for Db2 12): V13R1M500
- [Log files](https://ibm.github.io/db2forzosdeveloperextension-about/docs/troubleshooting/opening-an-issue-and-providing-logs) attached?: No

## Problem description
Selecting EBCDIC graphic data returns this error
`	[jcc][t4][1065][12306][4.29.24] Caught java.io.CharConversionException. See attached Throwable for details. ERRORCODE=-4220, SQLSTATE=null`

Selecting UNICODE graphic data is fine.
<img width="328" alt="image" src="https://user-images.githubusercontent.com/104032169/164885459-b627b5ee-e23b-4778-ba98-65e323a7d1e0.png">

Detailed steps for reproducing the problem:

1. Create a table with an EBCDIC graphic column
2. Populate the table
3. Select the EBCDIC graphic column with the Db2 Developer Extension

### Observed behavior

- ERRORCODE=-4220

### Expected behavior

- Success with no error

**Comment by katsoohoo** [comment]
Hi @pymak7,

Thank you for reporting this issue. We will take a look at it.

**Comment by katsoohoo** [comment]
Hi @pymak7,

We believe this issue may be occurring because the GRAPHIC or VARGRAPHIC input values are invalid when inserting into the table. When this occurs, Db2 will convert the invalid characters into X'FE'. Upon querying the table, the retrieved values cannot be converted to something readable so with hit the error. We were able to recreate the ERRORCODE=-4220 with this scenario.

Can you verify on your side if the valid values are being inserted into GRAPHIC or VARGRAPHIC columns?

**Comment by pymak7** [comment]
@katsoohoo 

Below is what I get when I query my EBCDIC graphic data in TEP3
```
SELECT V_RECID, V_GRAP7, HEX(V_GRAP7) FROM VWE1_ALLTYPE;                                                                      
                                                                                                                              
                                                                                                                              
                                                                                                                              
   +-----------------------------------------------------------------------------------------------------------------------+  
   | V_RECID |               V_GRAP7                |                                                                      |  
   +-----------------------------------------------------------------------------------------------------------------------+  
 1_| 111     | ..B.B.C.A.B.A                      . | 42C242C242C342C142C242C140404040404040404040404040404040404040404040 |  
 2_| 112     | ?                                    | ?                                                                    |  
 3_| 113     | ..A.B.C.A.B...A                    . | 42C142C242C342C142C2FEFE42C14040404040404040404040404040404040404040 |  
 4_| 114     | ..C.B.C.A.B.A                      . | 42C342C242C342C142C242C140404040404040404040404040404040404040404040 |  
 5_| 115     | ..D.B.C.A.B.A                      . | 42C442C242C342C142C242C140404040404040404040404040404040404040404040 |  
 6_| 116     | ..E.B.C.A.B.A                      . | 42C542C242C342C142C242C140404040404040404040404040404040404040404040 |  
 7_| 117     | ..F.B.C.A.B.A                      . | 42C642C242C342C142C242C140404040404040404040404040404040404040404040 |  
 8_| 118     | ..G.B.C.A.B.A                      . | 42C742C242C342C142C242C140404040404040404040404040404040404040404040 |  
 9_| 119     | ..H.B.C.A.B.A                      . | 42C842C242C342C142C242C140404040404040404040404040404040404040404040 |  
10_| 120     | ..I.B.C.A.B.A                      . | 42C942C242C342C142C242C140404040404040404040404040404040404040404040 |  
11_| 121     | ..J.B.C.A.B.A                      . | 42D142C242C342C142C242C140404040404040404040404040404040404040404040 |  
12_| 122     | ..K.B.C.A.B.A                      . | 42D242C242C342C142C242C140404040404040404040404040404040404040404040 |  
13_| 123     | ..L.B.C.A.B.A                      . | 42D342C242C342C142C242C140404040404040404040404040404040404040404040 |  
14_| 124     | ..M.B.C.A.B.A                      . | 42D442C242C342C142C242C140404040404040404040404040404040404040404040 |  
15_| 125     | ..A.B.C.A.B.A                      . | 42C142C242C342C142C242C140404040404040404040404040404040404040404040 |  
16_| 126     | ..N.B.C.A.B.A                      . | 42D542C242C342C142C242C140404040404040404040404040404040404040404040 |  
17_| 127     | ..O.B.C.A.B.A                      . | 42D642C242C342C142C242C140404040404040404040404040404040404040404040 |  
```

and EBCDIC vargraphic data
```
SELECT V_RECID, V_VGRAP5, HEX(V_VGRAP5) FROM VWE1_ALLTYPE;                                                              
                                                                                                                        
                                                                                                                        
                                                                                                                        
         +-----------------------------------------------------------------------------------------------------------+  
         | V_RECID |             V_VGRAP5             |                                                              |  
         +-----------------------------------------------------------------------------------------------------------+  
       1_| 111     | ..A.B.C.1.                       | 42C142C242C342F1                                             |  
       2_| 112     | ..A.B.C.2.                       | 42C142C242C342F2                                             |  
       3_| 113     | ..A.B...C.3.                     | 42C142C2FEFE42C342F3                                         |  
       4_| 114     | ..A.B.C.4.                       | 42C142C242C342F4                                             |  
       5_| 115     | ..A.B.C.5.                       | 42C142C242C342F5                                             |  
       6_| 116     | ..A.B.C.6.                       | 42C142C242C342F6                                             |  
       7_| 117     | ..A.B.C.7.                       | 42C142C242C342F7                                             |  
       8_| 118     | ..A.B.C.8.                       | 42C142C242C342F8                                             |  
       9_| 119     | ..A.B.C.9.                       | 42C142C242C342F9                                             |  
      10_| 120     | ..A.B.C.0.                       | 42C142C242C342F0                                             |  
      11_| 121     | ..A.B.C.A.                       | 42C142C242C342C1                                             |  
      12_| 122     | ..A.B.C.!.                       | 42C142C242C3425A                                             |  
      13_| 123     | ..A.B.C.,.                       | 42C142C242C3426B                                             |  
      14_| 124     | ..A.B.C./.                       | 42C142C242C34261                                             |  
      15_| 125     | ..A.B.C.\.                       | 42C142C242C342E0                                             |  
      16_| 126     | ..A.B.C.&.                       | 42C142C242C34250                                             |  
      17_| 127     | ..A.B.C.(.                       | 42C142C242C3424D                                             |  
```

**Comment by katsoohoo** [comment]
Hi @pymak7,

An update on this issue: After examining the data more closely, I noticed the rows with `V_RECID = '113'` and `V_RECID = '158'` contains [characters that can not be converted as indicated by �](https://www.ibm.com/support/pages/sqlexception-message-caught-javaiocharconversionexception-and-errorcode-4220). With the SQL statements below, I'm able to query the data in the extension.
```
SELECT V_RECID, V_GRAP7 FROM VWE1_ALLTYPE WHERE V_RECID != '113' AND V_RECID != '158';                                                                      
SELECT V_RECID, V_VGRAP5 FROM VWE1_ALLTYPE WHERE V_RECID != '113' AND V_RECID != '158';
```

Or query with � visible:
```
SELECT V_RECID, CAST(V_GRAP7 AS GRAPHIC(17) CCSID UNICODE) FROM VWE1_ALLTYPE;                                                                      
SELECT V_RECID, CAST(V_VGRAP5 AS GRAPHIC(17) CCSID UNICODE) FROM VWE1_ALLTYPE;     
```

For the next point of action, I think it makes sense to show the rest of the result set even if some of the characters cannot be displayed properly. We will look into this.

**Comment by katsoohoo** [comment]
Hi @pymak7,

With v2.1.0 release, the extension will display � when characters cannot be converted in the SQL results by default. This behavior can be changed with setting `db2forzosdeveloperextension.db2sqlservice.sQL Results.characterConversion`.

![Screen Shot 2023-02-24 at 10 01 11](https://user-images.githubusercontent.com/5027555/221255426-b0d91c1f-da79-4a18-8676-6b35b5a3d1c5.png)

**Comment by katsoohoo** [comment]
I will close this issue since the bug was fixed in v2.1.0.
